Code/Tools/AWSNativeSDKInit/include/AWSNativeSDKInit/AWSNativeSDKInit.h

```diff
@@ -13,7 +13,7 @@
 #include <AzCore/Module/Environment.h>
 
 #if defined(PLATFORM_SUPPORTS_AWS_NATIVE_SDK)
-
+#include <AzCore/PlatformIncl.h>
```

Comment:
Nit: Should this header file be included inside or outside #if defined(PLATFORM_SUPPORTS_AWS_NATIVE_SDK)?

In Code/Tools/AWSNativeSDKInit/include/AWSNativeSDKInit/AWSMemoryInterface.h, seems like it's included outside #if defined(PLATFORM_SUPPORTS_AWS_NATIVE_SDK).

I'm curious about the reason to include this header as well.

Reply:
strictly speaking the header should be included inside #if, but we don't really have practical use case that AWSNativeSDK is in used but PLATFORM_SUPPORTS_AWS_NATIVE_SDK is not defined, if it happens then these files will fail the build as implementation requires to include aws header.

And the reason to include this header is to resolve conflict function name with windows header, o3de has undefined several conflict function or macros for different platforms. It is a common issue, and our problem refers to GetObject https://github.com/Tencent/rapidjson/issues/1448